src/scripts/eastern_kingdoms/tirisfal_glades/scarlet_monastery/instance_scarlet_monastery.cpp

```diff
@@ -100,8 +104,7 @@ struct instance_scarlet_monastery : ScriptedInstance
     uint64 m_uiVorrelGUID;
     uint64 m_uiDoorHighInquisitorGUID;
     uint64 m_uiChapelDoorGUID;
-    uint64 m_uiAshbringerWielderGUID;
-
+    Player* m_uiAshbringerWielderGUID;
```

Comment:
You should not store Player* ! Player can log out, and then pointer points to deleted memory (not nullptr, but deleted memory region, and using it will cause segmentation fault). Change it back to GUID and use sObjectAccessor.FindPlayer or GetMap()->FindPlayer to get the Player* from the GUID.